text/0000-singleton-removal.md

```diff
@@ -0,0 +1,69 @@
+- Feature Name: `singleton-remove`
+- Start Date: 2021-02-11
+- RFC PR: [RFC#11](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/11)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This change removes the `NetworkingManager.Singleton` static, as well as the majority of other static state in MLAPI. 
```

Comment:
So usability-wise, one still creates an empty object, then adds the NetworkingManager component to it, yes?

Looking at scale demo, most of the singleton accesses are on IsClient / IsServer.   So would is code...

```
public class ExitButtonScript : MonoBehaviour
{
    public void OnExitScene() {
        if(NetworkingManager.Singleton.IsClient)  {
            NetworkingManager.Singleton.StopClient();
        }
     }
}
```
....turn into...
```
public class ExitButtonScript : MonoBehaviour
{
    public void OnExitScene() {
        if(NetManager.IsClient)  {
            NetManager.StopClient();
        }
     }
}
```

Reply:
that's right--you'd have the NetworkingManager you wanted to interact with dragged in as serialized field.